examples/khr-3hv/full project/controllers/supervisorsManager/khr3hvEnv.py

```diff
@@ -0,0 +1,202 @@
+                       "Current z position": self.robot.getPosition()[2],
+                       "Globan distance walekd": GLOBAL_DISTANCE
+                       })           
+        
+        motorPos.extend([i for i in self.motorPositionArr])
+        # Shift the last 10 observations by one, on the time window of 5
+        self.obs = shift(self.obs, -self.numObs, cval=np.NaN)
+        self.obs[-self.numObs:] = motorPos
+
+        return np.array(self.obs)
+
+    def get_reward(self, action):
+        """
+        Calculate the reward of the agent. The reward award agent when moving forward on the z-axis.
+        Also, we are using a second term so the agent moves further away from his previous position.
+        :return: Reward value
+        :rtype: float
+
+        """
+        reward = 2.5 * self.robot.getPosition()[2] + self.robot.getPosition()[2] - self.prev_pos 
+        
+        if self.counterLoging % self.save_freq == 0 and self.train and not self.useRay:
+            wandb.log({"reward": reward,
+                       "reward-1term-weight-pos": 2.5 * self.robot.getPosition()[2],
+                        "reward-2term-diff-possition": self.robot.getPosition()[2] - self.prev_pos
+                    })
+
+        self.prev_pos = self.robot.getPosition()[2]
+        return reward 
+
+    def is_done(self):
+        """
+        This method checks the termination criteria for each episode. 
+        If the criteria are satisfied return True otherwise it returns False.
+        :return: Termination criteria
+        :rtype: Boolean
+
+        """
+        # Agent fall.
+        if self.robot.getPosition()[1] < 0.5: 
+            return True     
+        # Agent walked out of the box
+        if self.robot.getPosition()[2] > 15: 
+            return True  
+        return False
+
+    def get_default_observation(self):
+        """
+        Simple implementation returning the default observation which is a zero 
+        vector in the shape of the observation space.
+        :return: Starting observation zero vector
+        :rtype: numpy array
+        """
+        self.prev_pos = 0
+        self.obs = np.zeros(self.t * self.numObs)
+        self.distance_walk = 0 
+
+        return np.zeros(self.observation_space.shape[0])
+
+    def apply_action(self, action):
+        """
+        This method uses the action list provided, which contains the next action to be 
+        executed by the robot.It contains a float number denoting the action. 
+        The corresponding motorPosition value is applied at each motor.
+        :param action: The list that contains the action value
+        :type action: list of float
+        """
+        motorIndexes = [0, 1, 2, 3, 4, 5]
+
+        for i, ac in zip(motorIndexes, action):
+            self.motorPositionArr[i] += ac
+            self.motorList[i].setVelocity(self.motor_velocity)
+            self.motorList[i].setPosition(ac)
```

Comment:
```suggestion
        for i, ac in enumerate(motorIndexes, action):
            self.motorPositionArr[i] += ac*2.356
            self.motorList[i].setVelocity(self.motor_velocity)
            self.motorList[i].setPosition(ac)
```
`enumerate` allows us to loop over something and have an automatic counter.
`ac*2.356` is from the suggestion on _UserWarning: We recommend you to use a symmetric and normalized Box action space (range=[-1, 1])_